packages/dx-react-grid/docs/guides/custom-plugin-development.md

```diff
@@ -0,0 +1,687 @@
+
+How can we get values of a particular column? We set data items using the `rows` property of the [Grid](../reference/grid.md) plugin. This plugin exports the `rows` getter. We can use it in our case. However, we will need to pass data items to our UI plugin (`ToolbarFilter`). So, we will declare a new getter for this task and name it `toolbarFilterDataItems`. The value of this getter will be calculated based on the value of the `rows` getter. For this task we will need to use computed functions. These functions calculates values based on values of other getters. As you remember, we will need not only filter our grid, but display the filtered column name near the select box. Since we are going to do that in the `ToolbarFilter` plugin we will need to get the column name in the plugin. A new getter will help us do this. Let's name it `toolbarFilterColumnName`. Moreover, we will use this getter for calculating a value for `toolbarFilterDataItems`. Add a new calculated function named, for example, `filterDataItemsComputed` that accepts the `rows` and `toolbarFilterColumnName` getters and calculates items for our select box. Our code might be as follows:
+
+
+*file toolbar-filter-state.js:*
+```jsx
+...
+import {
+    ..., Getter
+} from '@devexpress/dx-react-core';
+
+const filterDataItemsComputed = ({rows, toolbarFilterColumnName}) => {
+  return rows.reduce((acc, row) => {
+    if(!acc.includes(row[toolbarFilterColumnName])) {
+      acc.push(row[toolbarFilterColumnName]);
+    }
+    return acc;
+  }, []);
+}
+
+export class ToolbarFilterState extends React.PureComponent {
+  ...
+  render() {
+    return (
+      <Plugin
+        name="ToolbarFilterState"
+      >
+        <Getter name="toolbarFilterColumnName" value={this.props.defaultColumnName} />
+        <Getter name="toolbarFilterDataItems" computed={filterDataItemsComputed} />
+      </Plugin>
+    );
+  }
+}
+```
+
+Note, that the getters order is very important. All computed getters that depends on value getters should be included after the value getters. That is why we added `toolbarFilterColumnName` before `toolbarFilterDataItems`.
+
+To get `toolbarFilterDataItems` in the `ToolbarFilter` plugin and render items, we need to use [TemplateConnector](../../../core/docs/reference/template-connector.md) as shown below:
+
+*toolbar-filter.js:*
+```jsx
+import {
+  ...,
+  TemplateConnector
+} from '@devexpress/dx-react-core';
+
+export class ToolbarFilter extends React.PureComponent {
+  render() {
+    return (
+      <Plugin
+        ...
+      >
+        <Template name="toolbarContent">
+          <TemplatePlaceholder />
+          <TemplateConnector>
+          {({
+              toolbarFilterDataItems
+            }) => (
+              <Select>
+                {toolbarFilterDataItems.map((item, index) =>
+                  <MenuItem key={index} value={item}>{item}</MenuItem>
```

Comment:
It's better to use `item` instead of `index` as a key

Reply:
I think it doesn't matter in this particular case.

Reply:
It's just a React best practice: using a loop index as a key is discouraged